<?xml version="1.0"?>
<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  <arg name="trajopt_description" default="trajopt_description"/>
  <arg name="plotting" default="true"/>
  <arg name="rviz" default="true"/>
  <arg name="ifopt" default="true"/>
  <arg name="debug" default="false"/>
  <arg name="testing" default="false"/>
  <arg name="method_id" default="1"/>
  <arg name="verbose" default="false"/>

  <!-- Load universal robot description format (URDF) -->
  <param name="$(arg robot_description)" command="$(find xacro)/xacro --inorder '$(find kortex_description)/robots/urdf/gen3_robotiq_2f_85_tesseract.xacro'" />

  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg robot_description)_semantic" textfile="$(find kortex_description)/robots/urdf/gen3_robotiq_2f_85_tesseract.srdf" />

  <group unless="$(arg testing)">
    <!-- If needed, broadcast static tf for robot root -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 world base_link 100" /> -->

    <node pkg="kortex_motion_planning" type="kortex_motion_planning_node" name="kortex_motion_planning_node" output="screen">
      <param name="plotting" type="bool" value="$(arg plotting)"/>
      <param name="rviz" type="bool" value="$(arg rviz)"/>
      <param name="ifopt" type="bool" value="$(arg ifopt)"/>
      <param name="debug" type="bool" value="$(arg debug)"/>
      <param name="method_id" type="int" value = "$(arg method_id)"/>
      <param name="verbose" type="bool" value = "$(arg verbose)"/>
    </node>

    <!-- Launch visualization -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="kortex_motion_planning_rviz"
        args="-d $(find kortex_motion_planning)/config/kortex_motion_planning.rviz" />


    <node name="kortex_motion_execution_node" pkg="kortex_motion_planning" type="kortex_motion_execution_node"  output="screen">
    </node>
  </group>

</launch>
